File: apps/api/internal/ai/catalog_test.go
package ai

import (
	"strings"
	"testing"
)

func TestEveryTaskHasDefaultByProvider(t *testing.T) {
	for _, task := range Tasks {
		if task.DefaultByProvider == nil {
			t.Errorf("task %q has nil DefaultByProvider", task.ID)
		}
		if len(task.DefaultByProvider) == 0 {
			t.Errorf("task %q has empty DefaultByProvider", task.ID)
		}
	}
}

func TestDefaultByProviderReferencesRealModels(t *testing.T) {
	for _, task := range Tasks {
		for provider, modelID := range task.DefaultByProvider {
			if GetModel(modelID) == nil {
				t.Errorf("task %q DefaultByProvider[%q] = %q, but GetModel returns nil",
					task.ID, provider, modelID)
			}
		}
	}
}

func TestDefaultByProviderHasAllFiveProviders(t *testing.T) {
	requiredProviders := []string{"copilot", "claude-cli", "opencode", "sdk", "gemini"}
	for _, task := range Tasks {
		for _, p := range requiredProviders {
			if _, ok := task.DefaultByProvider[p]; !ok {
				t.Errorf("task %q DefaultByProvider missing provider %q", task.ID, p)
			}
		}
	}
}

func TestOpencodeDefaultsUseGoPrefix(t *testing.T) {
	for _, task := range Tasks {
		modelID := task.DefaultByProvider["opencode"]
		if modelID == "" {
			t.Errorf("task %q has no opencode default", task.ID)
			continue
		}
		// The opencode onboarding provider must map to opencode-go: entries,
		// not opencode: (free tier).
		if !strings.HasPrefix(modelID, "opencode-go:") {
			t.Errorf("task %q opencode default = %q; want opencode-go: prefix", task.ID, modelID)
		}
	}
}

File: packages/shared/src/catalog.test.js
import { describe, expect, it } from 'vitest';
import { getDefaultForTask, TASKS } from './catalog.js';

describe('getDefaultForTask', () => {
  it('returns the opencode-go default for post_scoring with opencode provider', () => {
    expect(getDefaultForTask('post_scoring', 'opencode')).toBe('opencode-go:deepseek-v4-flash');
  });

  it('returns the copilot default for post_scoring with copilot provider', () => {
    expect(getDefaultForTask('post_scoring', 'copilot')).toBe('copilot:gpt-5-mini');
  });

  it('falls back to task default when provider is unknown', () => {
    expect(getDefaultForTask('post_scoring', 'unknown')).toBe('copilot:gpt-5-mini');
  });

  it('returns null when task does not exist', () => {
    expect(getDefaultForTask('missing-task', 'opencode')).toBeNull();
  });

  it('returns the claude-cli default for advisor_council', () => {
    expect(getDefaultForTask('advisor_council', 'claude-cli')).toBe('claude-cli:opus');
  });
});

describe('TASKS defaultByProvider', () => {
  it('every task has a non-empty defaultByProvider object', () => {
    for (const task of TASKS) {
      expect(task.defaultByProvider).toBeDefined();
      expect(Object.keys(task.defaultByProvider).length).toBeGreaterThan(0);
    }
  });

  it('every task has all five providers in defaultByProvider', () => {
    const requiredProviders = ['copilot', 'claude-cli', 'opencode', 'sdk', 'gemini'];
    for (const task of TASKS) {
      for (const provider of requiredProviders) {
        expect(task.defaultByProvider[provider], `task ${task.id} missing provider ${provider}`).toBeDefined();
      }
    }
  });
});
